.github/workflows/scripts/common.sh

```diff
@@ -0,0 +1,45 @@
+#!/bin/bash
+
+# The output appears in a dedicated emphasized GitHub step box.
+# Multiple lines are not supported.
+# Markdown is not supported.
+gh_log() {
+    local LEVEL="$1"
+    shift
+    echo "::$LEVEL::$*"
+}
+export -f gh_log
+
+# Sets a step output value.
+# Multiple lines are not supported: join with URL escaping.
+gh_output() {
+    local NAME="$1"
+    shift
+    echo "::set-output name=$NAME::$*"
+}
+export -f gh_output
+
+check_not_empty() {
+    for V in "$@"; do
+        typeset -n VAR="$V"
+        if [ -z "${VAR:-}" ]; then
+            gh_log error "Variable $V is not set or empty"
+            exit 1
+        fi
+    done
+}
+export -f check_not_empty
+
+check_not_empty GITHUB_STEP_SUMMARY
+
+# The output appears in the GitHub step summary box.
+# Multiple lines are supported.
+# Markdown is supported.
+gh_summary() {
+    if [ "$#" -eq 0 ]; then
+        cat
+    else
+        echo -e "$@"
+    fi >>"$GITHUB_STEP_SUMMARY"
```

Comment:
Should:

Could we simplify this into

```suggestion
    echo -e "${@}" >>"$GITHUB_STEP_SUMMARY"
```

If `${@}` is empty then this function should add nothing to the file, right? (I am wondering why we need this `cat` in line 40 🤔

Reply:
Cat is there to pass data from the pipe. This function can be called either with arguments:
```
gh_summary "Some" "summary"
```
or on a pipe:
```
gh_summary <<EOF
multiline
summary
EOF
```

Reply:
I think this can be simplified, but there is no need to do this in this PR. Would you only add a comment in the code why we have this `cat`? It was not clear for me when reading the code.